src/av2/datasets/motion_forecasting/eval/submission.py

```diff
@@ -0,0 +1,102 @@
+# <Copyright 2022, Argo AI, LLC. Released under the MIT license.>
+
+"""Classes and utilities used to build submissions for the AV2 motion forecasting challenge."""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Final, List, Tuple
+
+import numpy as np
+import pandas as pd
+
+from av2.utils.typing import NDArrayNumber
+
+TrackPredictions = NDArrayNumber  # Track predictions are expected to be of shape (*, NUM_PRED_TIMESTEPS, 2)
+ScenarioPredictions = Dict[str, TrackPredictions]  # Mapping from track ID to track predictions
+PredictionRow = Tuple[str, str, int, NDArrayNumber, NDArrayNumber]
+
+SUBMISSION_COL_NAMES: Final[List[str]] = [
+    "scenario_id",
+    "track_id",
+    "prediction_rank",
+    "predicted_trajectory_x",
+    "predicted_trajectory_y",
+]
+NUM_PRED_TIMESTEPS: Final[int] = 60
+
+
+@dataclass
+class ChallengeSubmission:
+    """Representation used to build submission for the AV2 motion forecasting challenge.
+
+    Args:
+        predictions: Container for all predictions to score - mapping from scenario ID to scenario-level predictions.
+    """
+
+    predictions: Dict[str, ScenarioPredictions]
+
+    def __post_init__(self) -> None:
+        """Validate that each of the submitted predictions has the appropriate shape.
+
+        Raises:
+            ValueError: If at least one submitted track prediction is not of shape (*, NUM_PRED_TIMESTEPS, 2).
+        """
+        for scenario_predictions in self.predictions.values():
+            for track_predictions in scenario_predictions.values():
+                if track_predictions.shape[-2:] != (NUM_PRED_TIMESTEPS, 2):
```

Comment:
I think it'd be nice to assign `track_predictions.shape[-2:]` to a descriptive variable name.